src/system/SystemLayer.h

```diff
@@ -116,6 +116,38 @@ class DLL_EXPORT Layer
      */
     virtual CHIP_ERROR StartTimer(Clock::Timeout aDelay, TimerCompleteCallback aComplete, void * aAppState) = 0;
 
+    /**
+     * @brief
+     *   This method extends the timer expiry to the provided aDelay. This method must be called while in the Matter context
+     *   (from the Matter event loop, or while holding the Matter stack lock).
+     *   aDelay is not added to the Remaining time of the timer. The finish line is pushed back to aDelay.
+     *
+     *   @note The goal of this method is that the timer remaining time cannot be shrunk and only extended to a new time
+     *         If the provided new Delay is smaller than the timer's remaining time, the timer is left untouched.
+     *         In the other case the method acts like StartTimer
+     *
+     *   @param[in]  aDelay             Time before this timer fires.
+     *   @param[in]  aComplete          A pointer to the function called when timer expires.
+     *   @param[in]  aAppState          A pointer to the application state object used when timer expires.
+     *
+     *   @return CHIP_NO_ERROR On success.
+     *   @return CHIP_ERROR_NO_MEMORY If a timer cannot be allocated.
+     *   @return Other Value indicating timer failed to start.
+     */
+    virtual CHIP_ERROR ExtendTimerTo(Clock::Timeout aDelay, TimerCompleteCallback aComplete, void * aAppState) = 0;
```

Comment:
Since you, anyway, implement this as restarting the timer if needed, wouldn't it be enough (and more generic) that the system abstraction provide sth like `Clock::Timestamp GetTimerWakeupTime(TimerCompleteCallback aComplete, void * aAppState)` which would allow you to implement `ExtendTimerTo` in the common code and would serve as `IsTimerActive` as well (by returning `Clock::Timestamp::max` or Optional's null value in the case the answer is 'no').

Reply:
So I thought you could implement `ExtendTimerTo` as:
```
   auto wakeupTime = GetTimerWakeupTime(aComplete, aAppState);
   if (wakeupTime != Clock::Timestamp::max() && wakeupTime < now + aDelay) {
     StartTimer(aDelay, aComplete, aAppState);
   }
```
Would something like that work in the same way as the current code?

Reply:
GetTimerWakeupTime is a method of `TimerList` and each SystemLayer implementation has its own list no? As you implement it, `ExtendTimerTo` could be common, but each SystemLayer would need to implement a method to leverage `GetTimerWakeupTime` on their timer list. And then this extra method would need to account for `mExpiredTimers` on the `LayerImplSelect` side.

I think it leads to a similar outcome

Reply:
Yes, that's correct. It's just a suggestion though, so feel free to keep the current version if it's already implemented.